src/Storages/StorageMergeTree.h

```diff
@@ -58,6 +58,8 @@ class StorageMergeTree final : public MergeTreeData
 
     bool supportsTransactions() const override { return true; }
 
+    bool isRemote([[ maybe_unused ]] ContextPtr local_context) const override { return local_context->getSettingsRef().allow_experimental_parallel_reading_from_replicas && !local_context->getClientInfo().collaborate_with_initiator; }
```

Comment:
I think we'd better have a separate flag `supportRemoteReading()` or something like this.
Will call it only inside InterpreterSelectQuery, and `isRemote` will mean only Distributed storage. It seem to be safer.

Reply:
Also, we won't need to specify `context`. Can check this condition on the single caller side in InterpreterSelectQuery

Reply:
Yes, this is an ugly solution and has to be redone.